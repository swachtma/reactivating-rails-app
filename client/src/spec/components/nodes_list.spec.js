/* global expect, jest */
import React from 'react'
import { shallow } from 'enzyme'

import configEnzyme, { expectXInY } from '../../test_helper'
import NodesList from '../../components/nodes_list'

let test_props = {
  nodes: [
    {
      id: 1892,
      chapter_id: 15,
      node_type: 'code_block',
      content: "``` javascript(path)\nimport React, { Component } from 'react';"
    },
    {
      id: 1893,
      chapter_id: 15,
      node_type: 'paragraph',
      content: 'Here is some text'
    }
  ]
}

describe('<NodesList />', () => {
  let nodes = ['#nodes-list', 'CodeExample', 'ReactDOMFragment']
  let w = shallow(<NodesList {...test_props} />)
  nodes.forEach(node => {
    it('renders ' + node, () => expectXInY(node, w))
  })

  it('renders according to snapshot', () => expect(w).toMatchSnapshot())
})
