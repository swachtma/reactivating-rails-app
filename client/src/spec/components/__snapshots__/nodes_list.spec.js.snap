// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NodesList /> renders according to snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NodesList
    nodes={
      Array [
        Object {
          "chapter_id": 15,
          "content": "\`\`\` javascript(path)
import React, { Component } from 'react';",
          "id": 1892,
          "node_type": "code_block",
        },
        Object {
          "chapter_id": 15,
          "content": "Here is some text",
          "id": 1893,
          "node_type": "paragraph",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <CodeExample
          node={
            Object {
              "chapter_id": 15,
              "content": "\`\`\` javascript(path)
import React, { Component } from 'react';",
              "id": 1892,
              "node_type": "code_block",
            }
          }
        />,
        <ReactDOMFragment
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p>Here is some text</p>
",
            }
          }
        />,
      ],
      "id": "nodes-list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "node_1892",
        "nodeType": "class",
        "props": Object {
          "node": Object {
            "chapter_id": 15,
            "content": "\`\`\` javascript(path)
import React, { Component } from 'react';",
            "id": 1892,
            "node_type": "code_block",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "node_1893",
        "nodeType": "class",
        "props": Object {
          "dangerouslySetInnerHTML": Object {
            "__html": "<p>Here is some text</p>
",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <CodeExample
            node={
              Object {
                "chapter_id": 15,
                "content": "\`\`\` javascript(path)
import React, { Component } from 'react';",
                "id": 1892,
                "node_type": "code_block",
              }
            }
          />,
          <ReactDOMFragment
            dangerouslySetInnerHTML={
              Object {
                "__html": "<p>Here is some text</p>
",
              }
            }
          />,
        ],
        "id": "nodes-list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "node_1892",
          "nodeType": "class",
          "props": Object {
            "node": Object {
              "chapter_id": 15,
              "content": "\`\`\` javascript(path)
import React, { Component } from 'react';",
              "id": 1892,
              "node_type": "code_block",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "node_1893",
          "nodeType": "class",
          "props": Object {
            "dangerouslySetInnerHTML": Object {
              "__html": "<p>Here is some text</p>
",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
